packages/cpp/src/browser/cpp-build-configurations.ts

```diff
@@ -270,9 +270,13 @@ export class CppBuildConfigurationManagerImpl implements CppBuildConfigurationMa
     }
 
     getActiveConfig(root?: string): CppBuildConfiguration | undefined {
-        // Get the active workspace root for the given uri, else for the first workspace root.
-        const workspaceRoot = root ? root : this.workspaceService.tryGetRoots()[0].uri;
-        return this.activeConfigurations.get(workspaceRoot);
+        if (!root) {
```

Comment:
@vince-fugnitto should not you use `root` if it is not `undefined`?

I think the issue was that `this.workspaceService.tryGetRoots()[0]` returns `undefined` and we try to call `.uri` on it.

Reply:
@akosyakov yes you're correct, I've updated the code again.